src/main/webapp/app/grading-system/grading-key-overview/grading-key-overview.component.ts

```diff
@@ -68,10 +69,12 @@ export class GradingKeyOverviewComponent implements OnInit {
                 this.noParticipationGrade = gradeSteps.noParticipationGrade;
                 if (gradeSteps.maxPoints !== undefined) {
                     if (!this.isExam) {
-                        // calculate course max points based on exercises
-                        const course = this.courseCalculationService.getCourse(this.courseId!);
-                        const maxPoints = this.courseCalculationService.calculateTotalScores(course!.exercises!, course!).get(ScoreType.REACHABLE_POINTS);
-                        this.gradingSystemService.setGradePoints(this.gradeSteps, maxPoints!);
+                        let maxPoints = 0;
+                        const scoresPerExerciseTypeForCourse = this.scoresStorageService.getStoredScoresPerExerciseType(this.courseId!);
+                        if (scoresPerExerciseTypeForCourse && scoresPerExerciseTypeForCourse.get(ExerciseTypeTOTAL.TOTAL)) {
+                            maxPoints = scoresPerExerciseTypeForCourse.get(ExerciseTypeTOTAL.TOTAL)![ScoreType.REACHABLE_POINTS];
```

Comment:
Is there a reason for using this enum instead of a specific entry?
Same question for occurrences in other files.

Reply:
This was a result of trying to squeeze totalScores into the scoresPerExerciseType in the `CourseForDashboardDTO`. I refactored it as you suggested (totalScores as an additional field in the CourseForDashboardDTO) and this is not necessary anymore.